app/controllers/users/notification_settings_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Users
+  class NotificationSettingsController < ApplicationController
+    before_action :raise_suspended
+    before_action :authenticate_user!
+    after_action :verify_authorized
+
+    ALLOWED_PARAMS = %i[id
+                        email_badge_notifications
+                        email_comment_notifications
+                        email_community_mod_newsletter
+                        email_connect_messages
+                        email_digest_periodic
+                        email_follower_notifications
+                        email_membership_newsletter
+                        email_mention_notifications
+                        email_newsletter
+                        email_tag_mod_newsletter
+                        email_unread_notifications
+                        mobile_comment_notifications
+                        mod_roundrobin_notifications
+                        reaction_notifications
+                        welcome_notifications].freeze
+
+    def update
+      authorize current_user, policy_class: UserPolicy
+
+      if current_user.notification_setting.update(users_notification_setting_params)
+        notice = "Your notification settings have been updated."
```

Comment:
I don't think the `notice` variable is necessary, I would combine the two lines into

```ruby
flash[:settings_notice] = "Your notification settings have been updated."
```

This may well be a holdover from old code you copy/pasted but since we're here already maybe we can clean up a little bit.